src/dgipy/vcf.py

```diff
@@ -0,0 +1,195 @@
+"""Provides methods for annotating VCF with DGIdb data"""
+
+import contextlib
+from itertools import groupby
+from pathlib import Path
+
+import pandas as pd
+import pysam
+import requests
+from tqdm import tqdm
+
+import dgipy
+
+
+# TODO: Probably need another class as a wrapper object rather than putting it all in a list
+# Class would have analogous display methods but also allow access to individual GeneResults
+class GeneResult:
+    """A gene result from original VCF
+    .. code-block:: python
+        import vcf
+        # Provide path to VCF file and specify chromosome
+        data = vcf.annotate('link/to/file',chr='chr#')
+    """
+
+    def __init__(self, data: list) -> None:
+        """Initialize a gene with a name, VCF records, and drug-gene interactions"""
+        # TODO: handle genes without names, 'novel transript'
+        try:
+            self.gene = data[0]["name"]
+        except:
+            self.gene = "None"
+
+        self.records = data
+        self.interactions = self.search_interactions(self.gene)
+
+        # TODO: handle app searches with blank lists [], currently FDA resource hangs for awhile?
+        if not list(self.interactions["drug"].values):
+            self.applications = "None"
+        else:
+            self.applications = self.search_applications(
+                list(self.interactions["drug"].values)
+            )
+
+        self.gene_info = self.grab_gene_info(self.gene)
+        self.categories = self.grab_categories(self.gene)
+
+    # TODO: These can probably just be simpilifed to direct assignment during __init__ but might be useful this way too
+    def search_interactions(self, gene: str) -> pd.DataFrame:
+        """Search drug-gene interactions for given gene in DGIdb
+
+        :param gene: the name of the gene
+        :return: Dataframe of drug-gene interactions
+        """
+        return dgipy.get_interactions(gene)
+
+    def search_applications(self, drugs: list) -> pd.DataFrame:
+        """Search for drug applications for interaction drugs in DGIdb
+
+        :param drugs: list of drugs from interaction results
+        :return: Dataframe of FDA applications
+        """
+        return dgipy.get_drug_applications(drugs)
+
+    def grab_gene_info(self, gene: str) -> pd.DataFrame:
+        """Grab gene information from DGIdb
+
+        :param gene: gene name
+        :return: Dataframe of gene information
+        """
+        return dgipy.get_gene(gene)
+
+    def grab_categories(self, gene: str) -> pd.DataFrame:
+        """Grab gene categories from DGIdb
+
+        :param gene: gene name
+        :return: Dataframe of gene categories
+        """
+        return dgipy.get_categories(gene)
```

Comment:
is there a reason why these are all defined as separate methods when they're just wrapping a single method call?

Reply:
I think I had thought it would be useful to be able to update the records stored within the class just by calling the instantiated class object (gene.grab_categories() etc). In its current form, there really isn't probably much need for this bc nothing is transforming the data in any way but idk if that is always the case.

Reply:
The danger with stubbing out stuff like that before you have a solid need for it is the exact need often ends up shifting, and then you have to change anyway -- I'd recommend just keeping it simple for now and then refactoring when necessary